File: 0125-valid-palindrome/0125-valid-palindrome.java


class Solution {
    public boolean isPalindrome(String s) {
        
        s = s.toLowerCase();

        int left = 0; 
        int right = s.length()-1;

        while(left < right){
            //skip the non-alphanumeric from left
            while(left < right &&  !Character.isLetterOrDigit(s.charAt(left))){
                left++;
            }
             // skip the non-alphanumeric from right       
            while(left < right && !Character.isLetterOrDigit(s.charAt(right))){
                right--;
            }

            //compare character 

            if(s.charAt(left) != s.charAt(right)){
                return false;
            }

            left++;
            right--;
        }

        return true;

    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
  const cleanedS = s.toLowerCase().replace(/[^a-z0-9]/g, '');

    let left =0;
    let right =cleanedS.length-1;

    while(left < right){

        if(cleanedS.charAt(left) !== cleanedS.charAt(right)){
            return false;
        }
        left++;
        right--;

    }
    return true;
    
};